shared/httpHandler/httpHandler.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @typedef {import('@aws-sdk/client-glue/dist-types/GlueClient').ClientDefaults[requestHandler]} RequestHandler
+ * @typedef {import('@smithy/protocol-http/dist-types').HttpRequest} HttpRequest
+ * @typedef {import('@smithy/protocol-http/dist-types').HttpResponse} HttpResponse
+ **/
+
+const { net } = require('electron');
+const { isEmpty } = require('lodash');
+const { requestTimeout } = require('./requestTimeout');
+const { queryBuilder } = require('./queryBuilder');
+
+/**
+ * @class
+ * @implements { RequestHandler }
+ */
+class HttpHandler {
+	constructor({ requestTimeout, logger } = {}) {
+		this.requestTimeout = requestTimeout;
+		this.logger = logger;
+	}
+
+	getAbortError = () => {
+		const abortError = new Error('Request aborted');
+		abortError.name = 'AbortError';
+		return abortError;
+	};
+
+	/**
+	 * @param { HttpRequest } request
+	 * @param { Object } [options]
+	 * @param { AbortSignal } [options.abortSignal]
+	 * @returns { Promise<{ response: HttpResponse }> }
+	 */
+	async handle(request, { abortSignal } = {}) {
+		if (abortSignal?.aborted) {
+			return Promise.reject(this.getAbortError());
```

Comment:
@chulanovskyi-bs under this async function context isn't just using
```
  throw this.getAbortError();
```
the same?

Reply:
Agree, those are equivalent